flight/Modules/Stabilization/rate.c

```diff
@@ -0,0 +1,94 @@
+/**
+******************************************************************************
+* @addtogroup TauLabsModules Tau Labs Modules
+* @{
+* @addtogroup StabilizationModule Stabilization Module
+* @{
+*
+* @file       rate.c
+* @author     Tau Labs, http://taulabs.org, Copyright (C) 2015
+* @author     AJ Christensen <aj@junglistheavy.industries>
+* @brief      Rate mode
+*
+* @see        The GNU Public License (GPL) Version 3
+*
+*****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+#include "openpilot.h"
+
+#include "pid.h"
+#include "stabilization.h"
+#include "stabilizationsettings.h"
+
+// MATH!!!!
+#include "misc_math.h"
+
+// TODO(fujin): Should these live in the header?
+//! Private variables
+static float rate_integral[MAX_AXES];
+
+// TODO(fujin): Is this useful for rate stabilization?
+//static float rate_decay = 0.991f;
+
+// TODO(fujin): This function probably recvs too many args
+int stabilization_rate(float gyro,
+                       float command,
+                       float *output,
+                       float dT,
+                       bool reinit,
+                       uint32_t axis,
+                       struct pid *pid,
+                       StabilizationSettingsData *settings)
+{
+  // TODO(fujin): Might as well implement rate gyro suppression while I'm here.
```

Comment:
@mlyle as mentioned on IRC I would believe that MWRate is a totally different PID controller as well as combined stick scaling as well as gyro suppression, I may be mistaken though. I would imagine that having gyro suppression for Rate would allow us to make use of lower maximum deg/s + extremely high expo to allow for fully gyro suppressed acrobatic maneuvers.